comictalker/comictalkerapi.py

```diff
@@ -16,23 +16,50 @@
 from __future__ import annotations
 
 import logging
+import sys
+from typing import NamedTuple
+
+if sys.version_info < (3, 10):
+    from importlib_metadata import EntryPoint, distributions
+else:
+    from importlib.metadata import EntryPoint, distributions
 
-import comictalker.talkers.comicvine
 from comictalker.talkerbase import ComicTalker, TalkerError
 
 logger = logging.getLogger(__name__)
 
 
-def get_comic_talker(source_name: str) -> type[ComicTalker]:
-    """Retrieve the available sources modules"""
-    sources = get_talkers()
-    if source_name not in sources:
-        raise TalkerError(source=source_name, code=4, desc="The talker does not exist")
+class Plugin(NamedTuple):
+    package: str
+    version: str
+    entry_point: EntryPoint
+
+
+def get_comic_talker(talker_name: str) -> type[ComicTalker]:
+    """Returns the requested talker module"""
+    talkers = get_talkers()
+    if talker_name not in talkers:
+        raise TalkerError(source=talker_name, code=4, desc="The talker does not exist")
+
+    return talkers[talker_name].entry_point.load()
+
+
+def get_talkers() -> dict[str, Plugin]:
+    """Returns all comic talker plugins (internal and external)"""
+    talkers: dict[str, Plugin] = {}
+    for dist in distributions():
+        eps = dist.entry_points
+
+        # perf: skip parsing `.metadata` (slow) if no entry points match
+        if not any(ep.group in ("comictagger_talkers") for ep in eps):
+            continue
```

Comment:
The only information pulled out of the `.metadata` is the version which isn't used so this can be removed.

Reply:
My thinking here is the version of the external plugins should be logged as CT version and talker plugins will be unbound. If not adding it to `Plugin` class then maybe a simple debug in `get_talkers` is a better idea? Probably want to put in a `logger.info` for the plugins found anyway? An alternative could be a class var (which seems to be needed for a display name, i.e. Comic Vine)?

Reply:
For now lets keep this all internal, anything relating to external talkers should happen later.